### PR TITLE
Package prefix on windows bug

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
In order to support shading we must derive the package prefix. However the algorithm will not work on windows because it prematurely exits because the package prefix is set to packageNameEnd and then checks if packagePrefix equals packageNameEnd and exits.

Modifications:
- Windows package name should look for a path separator '\' or use the libraryPathName if no path separator exist

Result:
Package prefix parsing on windows should work.

### CODE REVIEW

File: openssl-dynamic/src/main/c/jnilib.c [hunk 1]
```diff
@@ -233,19 +242,23 @@ static char* parsePackagePrefix(const char* libraryPathName, jint* status) {
         return NULL;
     }
 #ifdef WIN32
-    // on windows we don't have a lib prefix
```
Comment (normanmaurer) [inline openssl-dynamic/src/main/c/jnilib.c]: @Scottmitch please preserve this comment (as its still true).

File: openssl-dynamic/src/main/c/jnilib.c [hunk 2]
```diff
@@ -233,19 +242,23 @@ static char* parsePackagePrefix(const char* libraryPathName, jint* status) {
...
-    int incr = 0;
-    char* packagePrefix = packageNameEnd;
+    char* packagePrefix = netty_internal_tcnative_util_rstrchar(packageNameEnd, libraryPathName, '\\');
+    if (packagePrefix == NULL) {
+        // The string does not have to specify a path [1].
+        // [1] https://msdn.microsoft.com/en-us/library/windows/desktop/ms683200(v=vs.85).aspx
```
Comment (normanmaurer) [inline openssl-dynamic/src/main/c/jnilib.c]: windows... pff ;)

File: openssl-dynamic/src/main/c/jnilib.c (hunk 1 above)
Comment (Scottmitch) [inline openssl-dynamic/src/main/c/jnilib.c]: done (and enhanced the comment)

**Comment by Scottmitch** [comment]
master (4deefa35)

**Pull request closed without merging**